### ISSUE TITLE
Accessibility: Screen readers no longer accurately report current ctrl+l result

### ISSUE THREAD

**Issue by zersiax** [dendronhq/dendron#1940]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [x] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [x] Lookup.  Dendron's Lookup Command
- [x] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

Screen readers used to be able to relatively instantly report the current lookup result while a user was typing. This feature has seemingly become quite a bit chattier which causes irrelevant info to be announced, blocking out what actually needs to be reported.

## To Reproduce

1. With NVDA for Windows running, hit ctrl+l in a workspace.
2. Type several characters.
3. Observe screen reader output.

## Expected behavior
As the list filters, the currently selected autocomplete result should be announced and written to the appropriate aria-live region that VS Code uses for this purpose.

### What Happens Instead

Screen reader announces unnecessary output about the number of results that have been found and will often echo that the note does not exist, create?
Interestingly enough this only happens as the search string is altered; using up and down arrow keys to cycle through the current list works as expected. 

## Desktop (please complete the following information)

- OS: Windows 11
- Browser: N/A

## Additional context

The way this screen reader magic works is through the use of either aria-live or role=alert regions within the DOM. I do not know how the internal VS Code widgets handle this, though.
I don't recall if the amount of results was echoed before this started breaking; it may not have. 
In testing you may encounter the phrase "Unknown" or frequent mentions of the currently opened file name followed by "edit field" or some such; this is not a Dendron issue but rather an incompatibility between an unmodified copy of NVDA and VS code sticking to the ARIA spec.

**Comment by hikchoi** [comment]
Hey @zersiax thank you for the issue.

I don't have the setup to reproduce this issue but we'll keep track of this.
The change that we made to lookup recently is adding the capability for autocomplete, but extension authors have no access to the DOM in VSCode. I suspect this behavior is coming from VSCode itself, unfortunately.

**Comment by zersiax** [comment]
@isidorn do you know by any chance if there's extension docs for this kind of scenario, and if not, do you have some guidance on what might have happened here?f

**Comment by zersiax** [comment]
@hikchoi  I indeed doubt you can do all that much directly in the DOM, I'm going to naively assume there might be a way for extension authors to influence this to a degree.
I just wrote up the situation morer clearly and I think I see where the incongruity is coming from. When typing , before hitting tab to autocomplete, focus consistently lands, and stays, on the "create new" option. When tab is pressed, focus shifts to the first result, which makes me feel like this is by design. The confusion arises because of two things:

- Hitting Enter accepts a code suggestion in the VS Code Editor, but does not appear to do so in the lookup view; you have to hit tab instead, which makes sense coming from Bash but threw me off.
- Screen readers only focus on one thing at a time, so unless you go looking you won't really know what's in your list apart from this "create new" option which has focus until you tab to autocomplete. That's not really trivially fixable though, apart from perhaps adding a brief screen reader announcement which indicates this, e.g. "Press tab to trigger autocomplete". How to do this, though, I have no idea, I can't really find docs on how extension authors do this kind of thing, hoping @isidorn might know more :)

**Comment by hikchoi** [comment]
thanks for the extra context @zersiax. I can only imagine how frustrating this issue will be. I am not the best person to discuss about screen reader issues so I'll talk with the team and see if we can try to make some progress.

**Comment by isidorn** [comment]
Sorry but I do not have the full context.
@hikchoi what exactly happens when user presses `Ctrl + I` with your extension? Is there a webView, or are you a `CompletionItemProvider`?

**Comment by hikchoi** [comment]
Hey @isidorn. Thank you for chiming in. Lookup (ctrl/cmd + L) is a feature of Dendron that lets you search notes within our hierarchical note taking structure. (more on this here: https://wiki.dendron.so/notes/a7c3a810-28c8-4b47-96a6-8156b1524af3/ if you need more context).

It's using a quickpick to display the existing notes that are indexed.
Recently we introduced a tab autocomplete feature for this quickpick. This isn't using `CompletionItemProvider` as it's not interfacing with Intellisense. We are watching the quickpick and auto-completing on tab based on what has been indexed under the hood (@nickolay-kondratyev please correct me if I'm wrong here).

We are suspecting that the auto complete feature is interfering the quickpick and confusing the screen reader which is causing inaccurate reports.

As far as I'm aware, extension authors have no control over the DOM in vscode, so it is a bit hard to diagnose and fix this on our end. Can you point us to any relevant piece of code in vscode that is related to the screen reader output?

@zersiax, as an interim solution, we are considering introducing a way to disable the autocomplete feature as I imagine this issue would make it very hard for you to use the lookup feature.

Can you also confirm that reverting the Dendron version back to when autocomplete wasn't introduced solves this problem? You can install past versions of Dendron in the extension panel by clicking on the cog wheel and selecting a past version in `Install another version...`. We introduced the lookup tab autocomplete feature in `0.71.0` Please let us know.

Pinging @kevinslin and @jonathanyeung for visibility.

@isidorn please let us know if you need more information on this matter.

**Comment by isidorn** [comment]
@hikchoi if you are using the VS Code components you should be good.
However to investigate this further I suggest you simply try @zersiax steps and see what the screen reader is actually reading.
If you are on Win just install NVDA, if you are Mac just turn on VoiceOver with Cmd + F5, if you on Linux install Orca.
Setting up a screen reader should be < 5mins.

@zersiax do you see this behaviour using just some core components of VS Code, like F1 for command palette?

**Comment by zersiax** [comment]
@isidorn  no, I don't, but those seem to work a little differently.
From what I can determine, the lookup dialog focuses on the "create note" option while you're typing, I think this is done so you can just hit enter to make a note with the name of the currently entered input rather than committing an autocomplete result, as the lookup dialog is also used to create new notes. As in, you type your search result and either it focuses on an existing note, which means that note gets opened on return, or it doesn't, in which case focus lands on the "create new note" option after which a note with the currently entered input is created on return. That is how it used to work, in any case.
What seems to happen now is that you type a string and get focused on the "create note" option regardless of a match, until you hit tab, which commits an autocomplete the screen reader user is at that point not aware of, but I'm guessing here and I may very well be wrong. If you'd like to repro, all you need to do is install the extension, initialize a workspace, create like 2 or 3 notes and use the ctrl+l or cmd+l hotkey to find one of them with VoiceOver (mac) or NVDA/JAWS (Windows) running.
@hikchoi while offering an option to disable this behavior would be good for a short-term solution, I can't deny that the older way of doing things worked better, I think that should be a very short--term patch while we figure out something better at best, unless we find some kind of reason why the current implementation can't be made to work better. I'm sure autocomplete is a productivity gain for most users,and turning it off would exclude a potential target audience from that gain, as well as set a precedent for how we handle cases like this moving forward, which is probably not what we want.

**Comment by hikchoi** [comment]
@isidorn Thank you for the suggestion and tip. Will give it a try.

@zersiax I understand your concern with the short term solution. But we offer ways to disable a lot of our existing features to accomodate many usage patterns. This won't make the feature go away, but simply provide a safety net.

@nickolay-kondratyev pinging for visibility: does it make sense to have an option to disable the auto-complete feature through configuration?

**Comment by zersiax** [comment]
@hikchoi  this makes sense to me as a first step.
